wordpress-mobile/WordPress-Android
# [Jetpack Social] Post settings publicize skip connections integration

Part of #18571 

Should only be merged after merging https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2788/ and updating the FluxC version from `trunk`.

To test:
1 - Log in with a .com account;
2 - Enable JetpackSocialFeatureConfig feature flag;
3 - Make sure you have at least one social sharing account connected;
4 - Select a SH site with Jetpack plugin or Jetpack Social plugin installed;
5 - Open post list;
6 - Select a draft or create a new one;
7 - Tap the overflow menu and select "Post settings";
8 - Scroll down to the "Social Sharing" section;
9 - Observe the value in the switch view next to the social sharing account (e.g. enabled);
10 - Tap the switch to change it's value;
11 - Move back to posts list to trigger an update of the draft post;
12 - Open post settings and check "Social Sharing" section again: the switch value (enabled or disabled) should be the same as you left in step 10;
13 - Open post settings of the same post on web: the connection switch should be enabled or disabled according to what was changed in the Android client;
14 - Change the switch view on web and save draft;
15 - On Android client, move back to the posts list to refresh the posts data;
16 - Open post settings -> "Social Sharing" and check the switch: it should be the same as web from step 14;
17 - Repeat from step 4 with a .com site selected.

## Regression Notes
1. Potential unintended areas of impact
None (code behind feature flag)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
The `ViewModel` will be tested in another dedicated PR.

PR submission checklist:


UI Changes testing checklist:
